icu4c/source/common/udatamem.h

```diff
@@ -44,7 +44,7 @@ struct UDataMemory {
     int32_t           length;      /* Length of the data in bytes; -1 if unknown.     */
 };
 
-U_CFUNC UDataMemory *UDataMemory_createNewInstance(UErrorCode *pErr);
+U_CFUNC UDataMemory U_EXPORT2 *UDataMemory_createNewInstance(UErrorCode *pErr);
```

Comment:
RE: linker errors

This is going to sound really silly, but I think you need to move the `*`.

The data type returned by the function is `UDataMemory*`. When you add the `U_EXPORT2`, this ends up (unintentionally) changing things.

```suggestion
U_CFUNC UDataMemory* U_EXPORT2 UDataMemory_createNewInstance(UErrorCode *pErr);
```

Reply:
Thank you! This alone wasn't sufficient, but locally on Mac also changing `U_CFUNC` to `U_CAPI` was needed also. Let's see what happens in CI.